### ISSUE TITLE
Size of a CSR tensor must be of length 2

### ISSUE THREAD

**Issue by aocsa** [pytorch/pytorch#57211]
## Issue description

Size of a CSR tensor must be of length 2.  This is one of the invariance for the CSR layout, currently there are no checks for this case and size property for CSR tensors accepts len(size) > 2. See the following example:  

## Code example 

When using the current master:

```
>>> import torch 
>>> crow_indices = [0, 2, 4]
>>> col_indices = [0, 1, 0, 1]
>>> values = [1, 2, 3, 4]
>>> sparse = torch.sparse_csr_tensor(torch.tensor(crow_indices),
                                                      torch.tensor(col_indices),
                                                      torch.tensor(values),
                                                      size=(2, 10, 2))
tensor(crow_indices=tensor([0, 2, 4]),
       col_indices=tensor([0, 1, 0, 1]),
       values=tensor([1, 2, 3, 4]), size=(2, 10, 2), nnz=4,
       layout=torch.sparse_csr)
>>> sparse.to_dense()
tensor([[[1, 0],
         [2, 0],
         ... 
         [0, 0]],

        [[3, 0],
         ...
         [0, 0]]])

```
Note: 
The issue was mentioned here https://github.com/pytorch/pytorch/pull/56744#discussion_r622570160.

### CSR Invariants about sizes: 
> Issue https://github.com/pytorch/pytorch/issues/57211
- `len(size) == 2` 

> Checks between line 64-80: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/sparse/SparseCsrTensor.cpp
- `(crow_indices.dim == col_indices.dim == values.dim) == 1` 
- `len(crow_indices) == size[0] + 1`
- `len(col_indices) == len(values) == nnz`

### CSR Invariants about values
> Check https://github.com/pytorch/pytorch/blob/46a32e075c0bdb54e3e2a08a370d785ee5c5a6cd/aten/src/ATen/native/sparse/SparseCsrTensor.cpp#L60 
- `crow_indices[0] == 0`
 
> Issue https://github.com/pytorch/pytorch/issues/56856
- `col_indices[crow_indices[i]:crow_indices[i+1]]`
- `values[crow_indices[i]:crow_indices[i+1]]`
- `crow_indices[-1] == nnz`

cc @aocsa @nikitaved @pearu @mruberry

**Comment by aocsa** [comment]
Closed here https://github.com/pytorch/pytorch/pull/59010
